firmware/hw_layer/debounce.cpp

```diff
@@ -6,16 +6,60 @@
  * @author David Holdeman, (c) 2020
  */
 #include "debounce.h"
+#include "pin_repository.h"
+#include "engine_configuration.h"
+#include "hardware.h"
 
-void ButtonDebounce::init (int t, brain_pin_e p, pin_input_mode_e m) {
+ButtonDebounce* ButtonDebounce::s_firstDebounce = nullptr;
+
+void ButtonDebounce::init (int t, brain_pin_e *p, pin_input_mode_e *m) {
+    if (!initialized) {
+        ButtonDebounce *listItem = s_firstDebounce;
+        if (listItem == nullptr) {
+            s_firstDebounce = this;
+        } else {
+            while (listItem->nextDebounce != nullptr) {
+                listItem = listItem->nextDebounce;
+            }
+            listItem->nextDebounce = this;
+        }
+    }
     threshold = MS2NT(t);
     timeLast = 0;
     pin = p;
+    active_pin = *p;
+    mode = m;
+    active_mode = *m;
```

Comment:
can probably deduplicate a little bit by just calling updateConfiguration(), but not a big deal